ext/opentelemetry-ext-system-metrics/src/opentelemetry/ext/system_metrics/__init__.py

```diff
@@ -0,0 +1,198 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Instrument to report system (CPU, memory, network) and
+process (CPU, memory, garbage collection) metrics.
+
+Usage
+-----
+
+.. code:: python
+
+    from opentelemetry.ext.system_metrics import SystemMetrics
+    from opentelemetry.sdk.metrics.export import ConsoleMetricsExporter
+
+    exporter = ConsoleMetricsExporter()
+    SystemMetrics(exporter)
+
+    # metrics are collected asynchronously
+    input("...")
+
+API
+---
+"""
+
+import gc
+import os
+import typing
+
+import psutil
+
+from opentelemetry import metrics
+from opentelemetry.sdk.metrics.export import MetricsExporter
+from opentelemetry.sdk.metrics.export.controller import PushController
+
+
+class SystemMetrics:
+    def __init__(
+        self,
+        exporter: MetricsExporter,
+        interval: int = 30,
+        labels: typing.Optional[typing.Dict[str, str]] = None,
+    ):
+        self._labels = {} if labels is None else labels
+        self.meter = metrics.get_meter(__name__)
+        self.controller = PushController(
+            meter=self.meter, exporter=exporter, interval=interval
+        )
+        self._proc = psutil.Process(os.getpid())
+        self._system_memory_labels = {}
+        self._system_cpu_labels = {}
+        self._network_bytes_labels = {}
+        self._runtime_memory_labels = {}
+        self._runtime_gc_labels = {}
+        # create the label set for each observer once
+        for key, value in self._labels.items():
+            self._system_memory_labels[key] = value
+            self._system_cpu_labels[key] = value
+            self._network_bytes_labels[key] = value
+            self._runtime_memory_labels[key] = value
+            self._runtime_gc_labels[key] = value
+
+        self.meter.register_observer(
+            callback=self._get_system_memory,
+            name="system.mem",
+            description="System memory",
+            unit="bytes",
+            value_type=int,
+            label_keys=self._labels.keys(),
```

Comment:
I don't think you need to supply `label_keys`. Originally, these were served as "recommended keys", in which if the metric is recorded with labels that are not in `label_keys`, they would be dropped. Since you are adding the "type" label in each of the observes, they would theoretically be dropped. However, this functionality is going to be done in `views` (so `label_keys` here doesn't actually do anything). As well, from the [specs](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/metrics/api.md#metric-instruments) we are most likely going to be getting rid of this field anyways.

Reply:
Removed